framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -273,13 +273,17 @@ private void annotateArrayAsArray(AnnotatedArrayType result, NewArrayTree node,
     }
 
     /**
-     * The type of a NewClassTree is the type of the Identifier
-     * plus any explicit annotations (including polymorphic qualifiers)
-     * on the constructor.
+     * Creates an AnnotatedDeclaredType for the NewClassTree and adds one of:
+     * <ul>
+     *   <li>explicit annotations from the new class expression ({@code new @HERE MyClass()} ), or</li>
+     *   <li>explicit annotations from the declaration of the class ({@code @HERE class MyClass {}} ), or</li>
```

Comment:
````
AnnotatedDeclaredType type = f.fromNewClass(node);
````
`type` contains explicit annotations on new class expression or explicit annotations from the declaration of the class or no annotations.

````
ConstructorReturnUtil.keepOnlyExplicitConstructorAnnotations(f, type, ex);
````
Annotations are only added to `type` (from `ex`) not removed.  `keepOnlyExplicitConstructorAnnotations` is poorly named and incorrectly documented. The `keepOnly` really means `copyOnly`. I'll update the documentation and rename that method.